File: models/biometricData.js
class BiometricData {
    constructor(id, userId, dataType, value, timestamp, sourceDevice, consentRecordId) {
        this.id = id;
        this.userId = userId;
        this.dataType = dataType;
        this.value = value;
        this.timestamp = timestamp;
        this.sourceDevice = sourceDevice;
        this.consentRecordId = consentRecordId;
    }
}

module.exports = BiometricData;

File: Sources/BiometricData.swift
import Foundation

struct BiometricData: Identifiable {
    // Represents the flexible structure of a JSONB field.
    struct BiometricValue: Codable {
        let type: String
        let value: Double
        let unit: String
    }

    let id: UUID
    let userId: UUID
    let dataType: String
    let value: BiometricValue
    let timestamp: Date
    let sourceDevice: String?
    let consentRecordId: UUID
}
